EPOParser.py

```diff
@@ -4,115 +4,133 @@
+    return fmap(lambda el: ( alias, el.tag, getattr(el, info) ), get_element(tree, elementPath))
 
 # parse :: Tree, [[ xpath, attrib | text, String ]] -> [(alias, tag {} | String)]
 def parse(tree, fields):
-  return fcompose(
-    fcmap(lambda x: x[0]),
-    fcfilter(lambda x: bool(x)),
-    fcmap(lambda field: getMetadata(tree, field[0], field[1], field[2]))
-  )(fields)
+    return fcompose(
+        fcmap(lambda x: x[0]),
+        fcfilter(lambda x: bool(x)),
+        fcmap(lambda field: get_metadata(tree, field[0], field[1], field[2]))
+    )(fields)
 
 ##############
 # processing #
 ##############
 
 def process(fields, state, callback, xmlFile, archiveFile):
-  if not os.path.isfile(marker(xmlFile)):
-    zipfile.ZipFile(archiveFile, "r").extract(xmlFile)
-    
+    if not os.path.isfile(marker(xmlFile)):
+        zipfile.ZipFile(archiveFile, "r").extract(xmlFile)
+
     # add validation/testing
-    
-    metadata = parse( tree(xmlFile), fields() )
-    
+
+    metadata = parse(tree(xmlFile), fields())
+
     # send off to queue (handle failures and restarts), if success, write file...
     callback(metadata)
-    
+
     # gives us a durable state if the script breaks and reruns
-    completefile = open(marker(xmlFile),"w") 
+    completefile = open(marker(xmlFile), "w")
     completefile.write(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
 
-    # gives us a final built object for logging/reporting, 
+    # gives us a final built object for logging/reporting,
     # may not want if it is too large in memory!
     state[archiveFile] = metadata
-    
-    os.remove(xmlFile)  
+
+    os.remove(xmlFile)
 
 def traverse(file, fields, state, callback):
-  initialDirectory = os.getcwd()
-  workDirectory = file[1].replace('\\', '/')[1:]
-
-  #print('init ' + initialDirectory)
-  #print('work ' + workDirectory)
-
-  archiveFile = file[0]
-  xmlFile = file[0].replace('zip', 'xml')
- 
-  os.chdir(workDirectory)
-  #print(os.getcwd())
-  process(fields, state, callback, xmlFile, archiveFile)
-  os.chdir(initialDirectory) 
-
-def fields(): 
-  return [
-    ['//abstract', 'attrib', 'abstract'], 
-    ['//abstract', 'text', 'abstract'], 
-    ['//ep-patent-document', 'attrib', 'ep-patent-document'], 
-    ['//B001EP', 'text', 'nametocallit']
-  ]
+    initialDirectory = os.getcwd()
+    workDirectory = file[1].replace('\\', '/')[1:]
+
+    #print('init ' + initialDirectory)
+    #print('work ' + workDirectory)
+
+    archiveFile = file[0]
+    xmlFile = file[0].replace('zip', 'xml')
+
+    os.chdir(workDirectory)
+    #print(os.getcwd())
+    process(fields, state, callback, xmlFile, archiveFile)
+    os.chdir(initialDirectory)
+
+
+def fields():
+    return [['//abstract', 'attrib', 'abstract'],
+            ['//abstract', 'text', 'abstract'],
+            ['//ep-patent-document', 'attrib', 'ep-patent-document'],
+            ['//B001EP', 'text', 'nametocallit']]
+
 
 def run(path, fields, state, callbackEach, callbackAll):
```

Comment:
This is probably an opportunity for a more descriptive name. Maybe `parse_all()`? I'm not in love with that, but it seems like this is parsing all xml files found in the index.